### ISSUE TITLE
Able to Custom Layer's unit test

### ISSUE THREAD

**Issue by MokkeMeguru** [tensorflow/tensorflow#36272]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.1.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
I want to write a correct batch normalization in MultiGPU/TPU.
I wrote some [code](https://gist.github.com/MokkeMeguru/35af0c7ddba511f6a268e7c78fdba2d6), but I cannot do your unit test.

**Will this change the current api? How?**
Add any argument for add custom layer in the function [layer_test](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/testing_utils.py#L75-L78)

**Who will benefit with this feature?**
1. Someone wants to fix an official layer.
2. Someone wants to create some layers.
 
**Any Other info.**

**Comment by MokkeMeguru** [comment]
So, we can change the layer_test function as follows
```python

def layer_test(layer_cls, kwargs=None, input_shape=None, input_dtype=None,
               input_data=None, expected_output=None,
               expected_output_dtype=None, expected_output_shape=None,
               validate_training=True, adapt_data=None,
               custom_objects=None): # add here
 # ...
 recovered_model = keras.models.Model.from_config(model_config, custom_objects) # change here
```

**Comment by MokkeMeguru** [comment]
Usage:
```python
testing_utils.layer_test(
    SyncBatchNormalization,
    # keras.layers.BatchNormalization,
    kwargs={
      'momentum': 0.9,
      'epsilon': 0.1,
      'gamma_regularizer': keras.regularizers.l2(0.01),
      'beta_regularizer': keras.regularizers.l2(0.01)
    },
    input_shape=(3, 4, 2),
    custom_objects={'SyncBatchNormalization': SyncBatchNormalization})
```

**Comment by tomerk** [comment]
Hi Mokke,
Thanks for the PR! I have two thoughts about this:

1. layer_test is a totally internal method, we don't want external users relying on it. This change seems explicitly designed for external users of Keras implementing their own layers and trying to use layer_test.

2. The layer_test utility isn't an effective test of layer behavior, it just makes sure certain things don't crash. You would be better served writing your own set of unit tests that actually verify layer behavior (and that make sure training, evaluation, etc. work correctly).

That said, I'll add @yhliang2018 to see if she thinks that there are some sort of testing utilities we should be exposing to external users in Keras' API.

**Comment by tanzhenyu** [comment]
Hi @MokkeMeguru you could have another test_serialization unit test which can cover this, for example, see [this](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/losses_test.py#L166-L170)

Closing it for now. Let us know if you have other questions!
